crux/models/resource.py

```diff
@@ -65,107 +47,114 @@ def __init__(
-        return self._description
+        return self.object_model["description"]
+
+    @description.setter
+    def description(self, description):
+        self.object_model["description"] = description
 
     @property
     def media_type(self):
         """str: Gets the Resource Description."""
-        return self._media_type
+        return self.object_model["mediaType"]
 
     @property
     def dataset_id(self):
         """str: Gets the Dataset ID."""
-        return self._dataset_id
+        return self.object_model["datasetId"]
 
     @property
     def folder_id(self):
         """str: Gets the Folder ID."""
-        return self._folder_id
+        return self.object_model["folderId"]
 
     @property
     def storage_id(self):
         """str: Gets the Storage ID."""
-        return self._storage_id
+        return self.object_model["storageId"]
 
     @property
     def name(self):
         """str: Gets the Resource Name."""
-        return self._name
+        return self.object_model["name"]
+
+    @name.setter
+    def name(self, name):
+        self.object_model["name"] = name
 
     @property
     def config(self):
         """str: Gets the config."""
-        return self._config
+        return self.object_model["config"]
+
+    @config.setter
+    def config(self, config):
+        self.object_model["config"] = config
 
     @property
     def provenance(self):
         """str: Gets the Provenance."""
-        return self._provenance
+        return self.object_model["provenance"]
+
+    @provenance.setter
+    def provenance(self, provenance):
+        self.object_model["provenance"] = provenance
 
     @property
     def type(self):
         """str: Gets the Resource Type."""
-        return self._type
+        return self.object_model["type"]
 
     @property
     def tags(self):
         """:obj:`list` of :obj:`str`: Gets the Resource Tags."""
-        return self._tags
+        return self.object_model["tags"]
+
+    @tags.setter
+    def tags(self, tags):
+        self.object_model["tags"] = tags
 
     @property
     def labels(self):
         """dict: Gets the Resource labels."""
-        return self._labels
+        labels = {}
+        for label in self.object_model["labels"]:
+            labels[label["labelKey"]] = label["labelValue"]
+        return labels
 
     @property
     def as_of(self):
         """str: Gets the as_of."""
-        return self._as_of
+        return self.object_model["asOf"]
 
     @property
     def created_at(self):
         """str: Gets created_at."""
-        return self._created_at
+        return self.object_model["createdAt"]
 
     @property
     def modified_at(self):
         """str: Gets modified_at."""
-        return self._modified_at
+        return self.object_model["modifiedAt"]
 
     @property
     def size(self):
         """int: Gets the size."""
-        return int(self._size)
+        size = self.object_model["size"]
+        return int(size if size else 0)
```

Comment:
It the API returns `null`, it must mean that file hasn't been uploaded yet, so letting this return `None` would be correct. There is a difference a zero byte file and no file. If there are parts of this or other code bases that use something like `if file.size < foo` we should update them with a `None` check.